### PR TITLE
Add sidecar sample test

### PR DESCRIPTION
**Pull request by gaoning777**
This change is [Reviewable](https://reviewable.io/reviews/kubeflow/pipelines/2106)

### CODE REVIEW

File: test/sample_test.yaml [hunk 1]
```diff
@@ -73,6 +73,7 @@ spec:
               - dsl_static_type_checking
               - pipeline_transformers
               - secret
+              - sidecar
               - execution_order
               - imagepullsecrets
               #- sidecar
```
Comment (numerology) [inline test/sample_test.yaml]: Can we just uncomment this line? Thanks!

Comment (gaoning777) [reply test/sample_test.yaml]: For sure.

**Comment by gaoning777** [comment]
@numerology this one is ready to be merged. I've tried to run it couple of times and the sample tests consistently pass.

**Comment by gaoning777** [comment]
fixing https://github.com/kubeflow/pipelines/issues/1813

**Pull request merged**